rclcpp/src/rclcpp/callback_group.cpp

```diff
@@ -48,6 +48,9 @@ CallbackGroup::add_subscription(
       subscription_ptrs_.end(),
       [](rclcpp::SubscriptionBase::WeakPtr x) {return x.expired();}),
     subscription_ptrs_.end());
+  if (call_executor_function_ != nullptr) {
+    call_executor_function_();
+  }
```

Comment:
You could consider putting this into a private method, like `notify_of_callback_group_change_if_needed()`.